Fixed the android app crash

avipars opened the pull request:
In reference to issue #189 , I have prevented the crash and tested it on a Android 14 Galaxy S23 Device

avipars committed:
Fixed the crash

avipars commented:
I would love to put the android app in it's own repo and then get it published on f-droid, I think there are a lot of DIYers and makers that would find it useful there. 

Let me know what you think!

martinohanlon commented:
I won't be able to look at this until September. 

I'm not against putting it in fdroid. Why does it need a separate repo?

avipars commented:
> I won't be able to look at this until September.
> 
> I'm not against putting it in fdroid. Why does it need a separate repo?

I am not super familiar with getting the app on f-droid, IIRC it needs to build the apk itself before publishing (and updates). @izzysoft might be able to tell us if its possible to build for f-droid from the nested clients folder

IzzySoft commented:
Sorry @avipars – but I've absolutely no plan with apps built from Python. I know F-Droid can handle some with buildozer, but I was never involved with those.

avipars commented:
> Sorry @avipars – but I've absolutely no plan with apps built from Python. I know F-Droid can handle some with buildozer, but I was never involved with those.



The android app client is actually in java, and it sits in a subdirectory of the repo. 

Would it be easier to add to f-drokd if the android app was in a separate repository?

IzzySoft commented:
Should work as is – apologies, I've just looked at the languages indicated by Github. F-Droid recipe then would need `subdir: clients/android/app`. Question they'll raise: last release as well as last commit were more than 18 months ago, so point out its maintenance status then (I see it's still actively supported in the issues here; no more commits ≠ not maintained, could well just mean "mature and no changes needed" :wink:)

avipars commented:
The pull request is unrelated (itself) to the comment/mention of f-droid. 



Please review the code changes and if it is to your liking (and you can test it and build it yourself), consider merging it.

I changed, built, ran, and tested the android app code on an s23 running android 14 and the crash no longer occurs.

IzzySoft commented:
@avipars I guess you were addressing @martinohanlon here and not the F-Droid inclusion – but just in case, let me point out that I'm currently [not active at F-Droid](https://gitlab.com/fdroid/admin/-/issues/447), so I'm out here :wink:

avipars committed:
Update FUNDING.yml

clach04 commented on .github/FUNDING.yml:
```diff
@@ -1,3 +1,5 @@
-# These are supported funding model platforms
-
-github: [martinohanlon]
+github: avipars
```
This doesn't seem like a reasonable change to this PR

clach04 reviewed (changes requested):
I don't know the code base to offer constructive feedback. I've definitely seen issues like you described related to permission checks with later Android versions. Thanks for spending time investigating and sharing a fix 👍 

If you've decided to fork this I recommend reopening this PR (without the funding change) using a feature branch (rather than master). That would make it easier for @martinohanlon to review and approve this

avipars commented:
was a mistake, i had an automation that added funding.yml to all my public repos... as my pr hasn't been reviewed since opening initially... it's not likely that it would get merged whatsoever so I'm just gonna close the issue

avipars committed:
Update README.rst

clach04 commented:
Those CI hooks will get you every time 😆 

Thanks again for spending time on this, I'm optimistic that @martinohanlon will have time in the future. In the meantime I took a copy of your fork, sans the accidental change in https://github.com/clach04/BlueDot/tree/issue_189_android_crash - Ideally, I'll make time to test this out and update the commit message with a mention of #189 but if @martinohanlon sees it ahead before then he can simply click through https://github.com/clach04/BlueDot/compare/issue_189_android_crash?expand=1 to open a new PR and merge it with relatively little effort thanks to your existing work.

I also saw a neat image optimization PR you had that looked neat so I grabbed that too at https://github.com/clach04/BlueDot/tree/avipars_imgbot

I've no idea when I'll get a chance to play with this, my (self-inflicted) TODO list is stupidly long 🤣 but you've saved future me a bunch of time, thank you! 🙏

avipars commented:
I decided to re-do the pull request without FUNDING (I had a script that went through all my repos and added it... unfortunately it also added the file to forked repos) @clach04 

IMGBOT is a good idea, but it went through the Android App Icons as well, and it may cause some issues with re-building the app... if you can test it on Android Studio first, that would be best

avipars committed:
removed funding
